### PR TITLE
Potential segfault in the time crate

### PR DESCRIPTION
**Pull request by jhpratt**

### CODE REVIEW

File: crates/time/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+[advisory]
```
Comment (tarcieri) [inline crates/time/RUSTSEC-0000-0000.md:1]: Needs \`\`\`toml surrounding the "front matter" here

File: crates/time/RUSTSEC-0000-0000.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+package = "time"
+date = "2020-11-17"
+url = "https://github.com/time-rs/time/issues/293"
+categories = ["code-execution", "memory-corruption"]
+cvss = "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H/E:F/RL:O/RC:C"
+keywords = ["segfault"]
+aliases = ["CVE-2020-26235"]
+
+[affected]
+# any Unix-like OS
```
Comment (tarcieri) [inline crates/time/RUSTSEC-0000-0000.md:12]: We don't have a good way of expressing this right now. Perhaps we should ala `cfg(unix)`

File: crates/time/RUSTSEC-0000-0000.md (hunk 1 above)
Comment (jhpratt) [reply crates/time/RUSTSEC-0000-0000.md:2]: Fixed.

File: crates/time/RUSTSEC-0000-0000.md (hunk 2 above)
Comment (jhpratt) [reply crates/time/RUSTSEC-0000-0000.md:13]: Yeah, I didn't see any nice way to represent it. I _think_ I caught all of them.

File: crates/time/RUSTSEC-0000-0000.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    "Linux",
+    "Redox",
+    "Solaris",
+    "Android",
+    "iOS",
+    "MacOS",
+    "NetBSD",
+    "OpenBSD",
+    "Bitrig",
+    "FreeBSD",
```
Comment (tarcieri) [inline crates/time/RUSTSEC-0000-0000.md:24]: These follow the same notation they do in Rust target triples. You can view the mapping here:

https://docs.rs/platforms/1.0.3/src/platforms/target/os.rs.html#69-89

tl;dr: lower case

Comment (jhpratt) [reply crates/time/RUSTSEC-0000-0000.md]: Fixed.

File: crates/time/RUSTSEC-0000-0000.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    "time::OffsetDateTime::now_local" = ["< 0.2.23"],
+    "time::OffsetDateTime::try_now_local" = ["< 0.2.23"],
+}
+
+[versions]
+patched = [">= 0.2.23"]
+unaffected = ["< 0.2.7"]
+```
+
+### Impact
```
Comment (tarcieri) [inline crates/time/RUSTSEC-0000-0000.md:41]: The advisory title is missing here:

```suggestion
# Potential segfault in the time crate

### Impact
```

Comment (jhpratt) [reply crates/time/RUSTSEC-0000-0000.md]: Fixed. Also reduced the header from h3 to h2 for the rest.

**Comment by jhpratt** [comment]
Not exactly sure what the lint is about:

> error loading advisory DB repo from .: RustSec error: parse error: error parsing ./crates/time/RUSTSEC-0000-0000.md: parse error: expected a table key, found a newline at line 25 column 14

That location doesn't exist. Line 25 in its entirety is `]`.

File: crates/time/RUSTSEC-0000-0000.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    "freebsd",
+]
+functions = {
+    "time::UtcOffset::local_offset_at" = ["< 0.2.23"],
+    "time::UtcOffset::try_local_offset_at" = ["< 0.2.23"],
+    "time::UtcOffset::current_local_offset" = ["< 0.2.23"],
+    "time::UtcOffset::try_current_local_offset" = ["< 0.2.23"],
+    "time::OffsetDateTime::now_local" = ["< 0.2.23"],
+    "time::OffsetDateTime::try_now_local" = ["< 0.2.23"],
+}
```
Comment (tarcieri) [inline crates/time/RUSTSEC-0000-0000.md:33]: The TOML parser doesn't like this notation. You might need to do:

```
[affected.functions]
"time::UtcOffset..." = ...
```

Comment (jhpratt) [reply crates/time/RUSTSEC-0000-0000.md]: I would've sworn that syntax was valid, but who knows. Fixed :)

File: crates/time/RUSTSEC-0000-0000.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+```toml
+[advisory]
+id = "RUSTSEC-0000-0000"
+package = "time"
+date = "2020-11-18"
+url = "https://github.com/time-rs/time/issues/293"
+categories = ["code-execution", "memory-corruption"]
+cvss = "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H/E:F/RL:O/RC:C"
```
Comment (tarcieri) [inline crates/time/RUSTSEC-0000-0000.md:8]: Latest thing is it doesn't like your CVSS v3.1 string. It looks like the set of metrics is incorrect?

Comment (jhpratt) [reply crates/time/RUSTSEC-0000-0000.md:8]: I took the string from [this site](https://www.first.org/cvss/calculator/3.1#CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H/E:F/RL:O/RC:C) (settings are correct). I've never used CVSS, and just took what it output as source truth. Perhaps it doesn't handle the "temporal score"?

Comment (tarcieri) [reply crates/time/RUSTSEC-0000-0000.md:8]: Yeah, it only handles the Base Metric Group

Comment (jhpratt) [reply crates/time/RUSTSEC-0000-0000.md]: CI is finally passing!

**Comment by tarcieri** [comment]
Thanks! Assigned `RUSTSEC-2020-0071`

**Pull request merged**